internal/tenant/context.go

```diff
@@ -16,23 +14,24 @@ var ErrMissingTenant = fmt.Errorf("missing tenant")
 func FromContext(ctx context.Context) (*tenanttype.Tenant, error) {
 	tnt, ok := tenanttype.GetTenant(ctx)
 	if !ok {
-		if pprofMissingTenant != nil {
-			// We want to track every stack trace, so need a unique value for the event
-			eventValue := pprofUniqID.Add(1)
-
-			// skip stack for Add and this function (2).
-			pprofMissingTenant.Add(eventValue, 2)
-		}
-
 		return nil, ErrMissingTenant
 	}
 	return tnt, nil
 }
 
-var pprofUniqID atomic.Int64
-var pprofMissingTenant = func() *pprof.Profile {
-	if !enforcement.ShouldLogNoTenant() {
-		return nil
+// IDToString is a helper function that returns a printable string of the tenant
+// ID in the context. This is useful for logging.
+func IDToString(ctx context.Context) string {
```

Comment:
kinda weird we have pprof as a side-effect here. Proposal to make it seem less weird: `func Log(*trace.Trace)`. Then inside of the sharded searcher you can do `tenant.Log(tr)`